main/views.py

```diff
@@ -57,3 +66,66 @@ def profile_creation(request):
 
     provider_form = ProviderForm()
     return render(request, "main/profile_creation.html", {"form": provider_form})
+
+
+
+@login_required(login_url="/login/")
+def connect_to_calendar(request):
+    '''Displays the page to allow the user to connect to their google calendar'''
+    user = request.user
+    profile = ProviderProfile.objects.get(user=user)
+    if profile.google_calendar_connected :
+        messages.info(request, "you are connected to calendar")
+        return redirect("providerdashboard")
+    else :
+        if request.method == "POST":
+            return redirect("connect_google")
+        
+        return render(request ,"main/connect_to_calendar.html")
+    
+
+def connect_google(request):
+    '''Creates the authorization url which the user is redirected to to allow for the connection '''
+
+    os.environ["OAUTHLIB_INSECURE_TRANSPORT"] = "1" # tell google that no https , using http
+
+    flow = Flow.from_client_secrets_file( # load google auth clint credentials
+        'credentials.json',
+        scopes = ["https://www.googleapis.com/auth/calendar"],
+        redirect_uri = "http://127.0.0.1:8000/google/oauth2callback/"
```

Comment:
Instead of hardcoding the URL we can create an env variable to store base URL and use that variable here